Installing volttron-core on Python 3.11 fails

**Describe the bug**
Cannot install volttron on Python 3.11. 

**To Reproduce**
Steps to reproduce the behavior:
1. Ensure that you are using Python 3.11
2. Create and activate a virtual env
3. Run `pip install volttron`
4. See error

**Expected behavior**

```
    ZMQ version detected: 4.3.4
      ************************************************
      building 'zmq.backend.cython._device' extension
      creating build/temp.macosx-12.6-x86_64-cpython-311/zmq
      creating build/temp.macosx-12.6-x86_64-cpython-311/zmq/backend
      creating build/temp.macosx-12.6-x86_64-cpython-311/zmq/backend/cython
      clang -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -DHAVE_SYS_UN_H=1 -I/usr/local/Cellar/zeromq/4.3.4/include -I/usr/local/Cellar/libsodium/1.0.18_1/include -Izmq/utils -I/Users/boni407/temp/.venv/include -I/Users/boni407/.pyenv/versions/3.11.0/include/python3.11 -c zmq/backend/cython/_device.c -o build/temp.macosx-12.6-x86_64-cpython-311/zmq/backend/cython/_device.o
      zmq/backend/cython/_device.c:217:12: fatal error: 'longintrepr.h' file not found
        #include "longintrepr.h"
                 ^~~~~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pyzmq
Failed to build gevent pyzmq
ERROR: Could not build wheels for gevent, pyzmq, which is required to install pyproject.toml-based projects
```

**Operating System (please complete the following information):**
 - OS: Ubuntu 18.04
 - Python 3.11

**Additional context**

Until this issue is resolved, recommend updating the python version in pyproject.toml at https://github.com/eclipse-volttron/volttron-core/blob/develop/pyproject.toml#L27

--- a/.github/workflows/run-tests.yml
+++ b/.github/workflows/run-tests.yml
@@ -21,7 +21,7 @@ defaults:
 env:
   LANG: en_US.utf-8
   LC_ALL: en_US.utf-8
-  PYTHON_VERSION: '3.8'
+  PYTHON_VERSION: '3.10'
   PROJECT_NAME: volttron-core
 
 jobs:
@@ -29,8 +29,8 @@ jobs:
   run-tests:
     strategy:
       matrix:
-        os: ["ubuntu-20.04", "ubuntu-22.04"]
-        python: ["3.8", "3.9", "3.10"]
+        os: ["ubuntu-22.04"]
+        python: ["3.10", "3.11"]
 
     runs-on: ${{ matrix.os }}
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -13,9 +13,8 @@ classifiers = [
     "Intended Audience :: Education",
     "Intended Audience :: Developers",
     "Programming Language :: Python",
-    "Programming Language :: Python :: 3.8",
-    "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
     "Topic :: Software Development :: Libraries :: Python Modules"
 ]
 
@@ -25,9 +24,9 @@ packages = [
 
 [tool.poetry.dependencies]
 poetry = "^1.2.2"
-python = "^3.8"
-pyzmq = "^22.3.0"
-gevent = "^21.12.0"
+python = "^3.10"
+pyzmq = "^25.0.2"
+gevent = "^22.10.2"
 PyYAML = "^6.0"
 toml = "^0.10.2"
 dateutils = "^0.6.12"
@@ -37,7 +36,7 @@ cryptography = "^36.0.1"
 watchdog-gevent = "^0.1.1"
 pip = "22.2.2"
 
-[tool.poetry.dev-dependencies]
+[tool.poetry.group.dev.dependencies]
 pytest = "^6.2.5"
 mock = "^4.0.3"
 pre-commit = "^2.17.0"

